source/FreeRTOS_IP.c

```diff
@@ -1441,84 +1441,101 @@ BaseType_t xSendEventStructToIPTask( const IPStackEvent_t * pxEvent,
 
+        switch( pxEthernetHeader->usFrameType )
+        {
+            case ipARP_FRAME_TYPE:
+            case ipIPv4_FRAME_TYPE:
+                #if ipconfigIS_DISABLED( ipconfigUSE_IPv4 )
+                    eReturn = eReleaseBuffer;
+                #endif
+                break;
+
+            case ipIPv6_FRAME_TYPE:
+                #if ipconfigIS_DISABLED( ipconfigUSE_IPv6 )
+                    eReturn = eReleaseBuffer;
+                #endif
+                break;
+
+            default:
+                if( FreeRTOS_ntohs( pxEthernetHeader->usFrameType ) <= 0x600U )
+                {
+                    #if ipconfigIS_ENABLED( ipconfigFILTER_OUT_NON_ETHERNET_II_FRAMES )
+                        eReturn = eReleaseBuffer;
+                    #endif
+                }
+                else
+                {
+                    #if ipconfigIS_DISABLED( ipconfigPROCESS_CUSTOM_ETHERNET_FRAMES )
+                        eReturn = eReleaseBuffer;
+                    #endif
+                }
+                break;
+        }
+
         /* Examine the destination MAC from the Ethernet header to see if it matches
          * that of an end point managed by FreeRTOS+TCP. */
-        pxEndPoint = FreeRTOS_FindEndPointOnMAC( &( pxEthernetHeader->xDestinationAddress ), NULL );
-
+        pxEndPoint = FreeRTOS_MatchingEndpoint( NULL, pucEthernetBuffer );
         if( pxEndPoint != NULL )
         {
-            /* The packet was directed to this node - process it. */
-            eReturn = eProcessBuffer;
-        }
-        else if( memcmp( xBroadcastMACAddress.ucBytes, pxEthernetHeader->xDestinationAddress.ucBytes, sizeof( MACAddress_t ) ) == 0 )
-        {
-            /* The packet was a broadcast - process it. */
-            eReturn = eProcessBuffer;
-        }
-        else
-        #if ( ( ipconfigUSE_LLMNR == 1 ) && ( ipconfigUSE_DNS != 0 ) )
-            if( memcmp( xLLMNR_MacAddress.ucBytes, pxEthernetHeader->xDestinationAddress.ucBytes, sizeof( MACAddress_t ) ) == 0 )
-            {
-                /* The packet is a request for LLMNR - process it. */
-                eReturn = eProcessBuffer;
-            }
-            else
-        #endif /* ipconfigUSE_LLMNR */
-        #if ( ( ipconfigUSE_MDNS == 1 ) && ( ipconfigUSE_DNS != 0 ) )
-            if( memcmp( xMDNS_MacAddress.ucBytes, pxEthernetHeader->xDestinationAddress.ucBytes, sizeof( MACAddress_t ) ) == 0 )
-            {
-                /* The packet is a request for MDNS - process it. */
-                eReturn = eProcessBuffer;
-            }
-            else
-        #endif /* ipconfigUSE_MDNS */
-        if( ( pxEthernetHeader->xDestinationAddress.ucBytes[ 0 ] == ipMULTICAST_MAC_ADDRESS_IPv6_0 ) &&
-            ( pxEthernetHeader->xDestinationAddress.ucBytes[ 1 ] == ipMULTICAST_MAC_ADDRESS_IPv6_1 ) )
-        {
-            /* The packet is a request for LLMNR - process it. */
-            eReturn = eProcessBuffer;
+            break;
         }
         else
```

Comment:
Starting an else clause is not necessary here:
~~~diff
        pxEndPoint = FreeRTOS_MatchingEndpoint( NULL, pucEthernetBuffer );
        if( pxEndPoint != NULL )
        {
            break;
        }
-       else
-       {
        #if ipconfigIS_ENABLED( ipconfigUSE_DNS )
            #if ipconfigIS_ENABLED( ipconfigUSE_LLMNR )
                #if ipconfigIS_ENABLED( ipconfigUSE_IPv4 )
                    if( memcmp( xLLMNR_MacAddress.ucBytes, pxEthernetHeader->xDestinationAddress.ucBytes, sizeof( MACAddress_t ) ) == 0 )
~~~